File: analysis/import/SonarImporter/src/main/java/de/maibornwolff/codecharta/importer/sonar/model/ErrorResponse.java
package de.maibornwolff.codecharta.importer.sonar.model;

import java.util.Arrays;

public class ErrorResponse {

    private ErrorEntity[] errors;

    public ErrorResponse() {
    }

    public ErrorResponse(ErrorEntity[] errors) {
        this.errors = errors;
    }

    public ErrorEntity[] getErrors() {
        return errors;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        ErrorResponse that = (ErrorResponse) o;

        // Probably incorrect - comparing Object[] arrays with Arrays.equals
        return Arrays.equals(errors, that.errors);
    }

    @Override
    public int hashCode() {
        return Arrays.hashCode(errors);
    }
}

File: analysis/import/SonarImporter/src/main/kotlin/de/maibornwolff/codecharta/importer/sonar/model/ErrorResponse.kt
package de.maibornwolff.codecharta.importer.sonar.model

import java.util.Arrays

data class ErrorResponse(val errors: Array<ErrorEntity>) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ErrorResponse

        if (!Arrays.equals(errors, other.errors)) return false

        return true
    }

    override fun hashCode(): Int {
        return Arrays.hashCode(errors)
    }
}
